--- a/arcade_galaga_basys3.srcs/sources_1/imports/rtl_dar/galaga.vhd
+++ b/arcade_galaga_basys3.srcs/sources_1/imports/rtl_dar/galaga.vhd
@@ -127,7 +127,9 @@
  start2         : in std_logic;
  left2          : in std_logic;
  right2         : in std_logic;
- fire2          : in std_logic
+ fire2          : in std_logic;
+ dip_switch_a   : in std_logic_vector(7 downto 0);
+ dip_switch_b   : in std_logic_vector(7 downto 0)
  );
 end galaga;
 
@@ -229,8 +231,6 @@
 
  signal cs05XX_ctrl       : std_logic_vector( 5 downto 0);
  
- signal dip_switch_a  : std_logic_vector (7 downto 0);
- signal dip_switch_b  : std_logic_vector (7 downto 0);
  signal dip_switch_do : std_logic_vector (1 downto 0);
  
  signal bgtile_num     : std_logic_vector( 7 downto 0);
@@ -321,8 +321,6 @@
 clock_18n <= not clock_18;
 reset_n   <= not reset;
 
-dip_switch_a <= "11110111"; --  cab:7 / na:6 / test:5 / freeze:4 / demo sound:3 / na:2 / difficulty:1-0
-dip_switch_b <= "10010111"; --lives:7-6/ bonus:5-3 / coinage:2-0
 dip_switch_do <= 	dip_switch_a(to_integer(unsigned(mux_addr(3 downto 0)))) & 
 									dip_switch_b(to_integer(unsigned(mux_addr(3 downto 0))));
